homeassistant/helpers/template.py

```diff
@@ -141,6 +145,34 @@ def async_render(self, variables: TemplateVarsType = None,
         except jinja2.TemplateError as err:
             raise TemplateError(err)
 
+    def render_with_collect(
+            self, variables: TemplateVarsType = None,
+            **kwargs) -> (Union[str, TemplateError],
```

Comment:
Let's not add a sync version. All template related stuff should be done in async. The old one had one because of backwards compat/tests.

Reply:
I agree, I will change the test cases and clean up the sync path in a separate PR though. I'm not confident that we can do that just yet.